controlplane/kubeadm/controllers/kubeadm_control_plane_controller_test.go

```diff
@@ -421,13 +421,10 @@ func TestReconcileNoClusterOwnerRef(t *testing.T) {
 		Log:    log.Log,
 	}
 
-	result, err := r.reconcile(context.Background(), kcp, r.Log)
+	result, err := r.Reconcile(ctrl.Request{NamespacedName: types.NamespacedName{Name: kcp.Name, Namespace: kcp.Namespace}})
 	g.Expect(err).NotTo(gomega.HaveOccurred())
 	g.Expect(result).To(gomega.Equal(ctrl.Result{}))
 
-	// Always expect that the Finalizer is set on the passed in resource
-	g.Expect(kcp.Finalizers).To(gomega.ContainElement(controlplanev1.KubeadmControlPlaneFinalizer))
```

Comment:
If we can't find the cluster, the Reconcile function now returns early, and the finalizer isn't set. This is in line with what the other controllers are doing today